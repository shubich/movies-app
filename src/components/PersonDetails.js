import React from 'react';
import ReactLoading from 'react-loading';
import { Poster } from './Poster';

export class PersonDetails extends React.Component {
    constructor(props) {
        super(props);
        if (!this.props.fetching) {
            this.props.getPersonDetails();            
            this.props.getFilmsWithCast();
            this.props.setSearchType();            
        }
    }
    
    render() {
        let details = this.props.details;
        
        return (
            (details.name || this.props.fetching) ?
            <div>
                {
                    this.props.fetching ?
                    <ReactLoading type='bars' delay={0} />
                    :
                    <div id='details'>
                        <Poster path={details.profile_path} />
                        <div className='info'>
                            <h2>
                                <span className='title text-primary'>{details.name}</span>
                                <span className='rating'>{details.popularity}</span>
                            </h2>
                            <p className='numbers'>
                                {details.born}
                            </p>
                            <p className='numbers'>
                                {details.died}
                            </p>
                            <p className='text-basic'>{details.biography}</p>
                        </div>
                    </div>
                }

            </div>
            :
            null
        )
    }
}