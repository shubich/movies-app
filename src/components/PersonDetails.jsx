import React from 'react';
import PropTypes from 'prop-types';
import ReactLoading from 'react-loading';
import Poster from './Poster';

export default class PersonDetails extends React.Component {
  componentDidMount() {
    const personId = this.props.match.params.id;

    if (!this.props.fetching) {
      this.props.getPersonDetailsAsync(personId);
      this.props.getFilmsWithCastAsync(personId);
      this.props.setSearchType('title');
    }
  }

  render() {
    return (
      (this.props.details.name || this.props.fetching) ?
        <div>
          {
            this.props.fetching ?
              <ReactLoading type="bars" delay={0} />
            :
              <div id="details">
                <Poster path={this.props.details.profile_path} />
                <div className="info">
                  <h2>
                    <span className="title text-primary">{this.props.details.name}</span>
                    <span className="rating">{this.props.details.popularity}</span>
                  </h2>
                  <p className="numbers">
                    {this.props.details.birthday + this.props.details.place_of_birth}
                  </p>
                  <p className="numbers">
                    {this.props.details.deathday}
                  </p>
                  <p className="text-basic">{this.props.details.biography}</p>
                </div>
              </div>
          }

        </div>
        :
        null
    );
  }
}

PersonDetails.propTypes = {
  fetching: PropTypes.bool.isRequired,
  getPersonDetailsAsync: PropTypes.func.isRequired,
  getFilmsWithCastAsync: PropTypes.func.isRequired,
  setSearchType: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
  details: PropTypes.shape({
    profile_path: PropTypes.string,
    name: PropTypes.string,
    birthday: PropTypes.string,
    place_of_birth: PropTypes.string,
    deathday: PropTypes.string,
    biography: PropTypes.string,
    popularity: PropTypes.number,
  }).isRequired,
};
