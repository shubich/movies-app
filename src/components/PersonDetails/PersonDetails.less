@import "../../styles/palette";

.PersonDetails {
    margin-top: 40px;
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-gap: 15px;
}

.PersonDetails-Poster {
    width: 250px;
}


.PersonDetails-Title {
    color: @primary-color;
}

.PersonDetails-Rating {
    color: @primary-color;
    margin-left: 25px;
    font-size: 0.7em;
    font-weight: 100;
    border: 1px solid white;
    background: white;
    border-radius: 50px;
    padding: 5px;
    font-family: initial;
    display: inline-block;
    min-width: 20px;
    text-align: center;
}

.PersonDetails-Date {
    font-weight: bold;
}

.PersonDetails-Biography {
    color: @basic-color;
}