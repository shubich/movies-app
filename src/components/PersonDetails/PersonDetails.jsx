import React from 'react';
import PropTypes from 'prop-types';
import Poster from '../Poster';
import Details from '../Details';

import './PersonDetails.less';

const PersonDetails = (props) => {
  if (!props.details.name) return null;

  return (
    <Details fetching={props.fetching}>
      <Poster
        className="PersonDetails-Poster Details-Poster"
        path={props.details.profile_path}
        person
      />
      <div className="PersonDetails-Info">
        <h2 className="PersonDetails-Header Details-Header">
          <span className="PersonDetails-Title Details-Title">
            {props.details.name}
          </span>
          {
            props.details.popularity ? (
              <span className="PersonDetails-Rating Details-Rating">
                {props.details.popularity.toFixed(0)}
              </span>
            ) : (null)
          }
        </h2>
        <p className="PersonDetails-Date">
          {
            props.details.birthday ? (
              `Born: ${
              new Date(props.details.birthday)
              .toLocaleDateString('en-US', {
                year: 'numeric', month: 'short', day: 'numeric',
              })}`
            ) : (null)
          }
          {
            props.details.place_of_birth ? (
              ` in ${props.details.place_of_birth}`
            ) : (null)
          }
        </p>
        <p className="PersonDetails-Date">
          {
            props.details.deathday ? (
              `Died: ${
              new Date(props.details.deathday)
              .toLocaleDateString('en-US', {
                year: 'numeric', month: 'short', day: 'numeric',
              })}`
            ) : (null)
          }
        </p>
        <p className="PersonDetails-Biography">
          {props.details.biography}
        </p>
      </div>
    </Details>
  );
};

PersonDetails.propTypes = {
  fetching: PropTypes.bool.isRequired,
  details: PropTypes.shape({
    profile_path: PropTypes.string,
    name: PropTypes.string,
    birthday: PropTypes.string,
    place_of_birth: PropTypes.string,
    deathday: PropTypes.string,
    biography: PropTypes.string,
    popularity: PropTypes.number,
  }).isRequired,
};

export default PersonDetails;
