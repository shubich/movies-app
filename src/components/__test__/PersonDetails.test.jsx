import React from 'react';
import { shallow } from 'enzyme';
import { PersonDetails } from '../PersonDetails';

let props;

describe('PersonDetails', () => {
  beforeEach(() => {
    props = {
      details: {
        name: 'Pulp Fiction',
        profile_path: '/img1234_0987',
        popularity: '1999',
        birthday: '8.3',
        place_of_birth: ['Thriller', 'Crime'],
        deathday: '154',
        biography: 'Lorem ipsum dolor sit amet.',
      },
      getPersonDetails: jest.fn(),
      getFilmsWithCast: jest.fn(),
      setSearchType: jest.fn(),
    };
  });

  it('renders correctly with details', () => {
    const wrapper = shallow(<PersonDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with fetching', () => {
    const wrapper = shallow(<PersonDetails {...props} fetching />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without details', () => {
    const wrapper = shallow(<PersonDetails {...props} details={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
