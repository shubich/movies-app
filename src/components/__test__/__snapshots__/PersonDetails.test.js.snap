// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetails renders correctly with details 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonDetails
    details={
      Object {
        "biography": "Lorem ipsum dolor sit amet.",
        "birthday": "8.3",
        "deathday": "154",
        "name": "Pulp Fiction",
        "place_of_birth": Array [
          "Thriller",
          "Crime",
        ],
        "popularity": "1999",
        "profile_path": "/img1234_0987",
      }
    }
    getFilmsWithCast={[Function]}
    getPersonDetails={[Function]}
    setSearchType={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="details"
      >
        <Poster
          path="/img1234_0987"
        />
        <div
          className="info"
        >
          <h2>
            <span
              className="title text-primary"
            >
              Pulp Fiction
            </span>
            <span
              className="rating"
            >
              1999
            </span>
          </h2>
          <p
            className="numbers"
          >
            8.3Thriller,Crime
          </p>
          <p
            className="numbers"
          >
            154
          </p>
          <p
            className="text-basic"
          >
            Lorem ipsum dolor sit amet.
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Poster
            path="/img1234_0987"
          />,
          <div
            className="info"
          >
            <h2>
              <span
                className="title text-primary"
              >
                Pulp Fiction
              </span>
              <span
                className="rating"
              >
                1999
              </span>
            </h2>
            <p
              className="numbers"
            >
              8.3Thriller,Crime
            </p>
            <p
              className="numbers"
            >
              154
            </p>
            <p
              className="text-basic"
            >
              Lorem ipsum dolor sit amet.
            </p>
          </div>,
        ],
        "id": "details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "path": "/img1234_0987",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                <span
                  className="title text-primary"
                >
                  Pulp Fiction
                </span>
                <span
                  className="rating"
                >
                  1999
                </span>
              </h2>,
              <p
                className="numbers"
              >
                8.3Thriller,Crime
              </p>,
              <p
                className="numbers"
              >
                154
              </p>,
              <p
                className="text-basic"
              >
                Lorem ipsum dolor sit amet.
              </p>,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="title text-primary"
                  >
                    Pulp Fiction
                  </span>,
                  <span
                    className="rating"
                  >
                    1999
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pulp Fiction",
                    "className": "title text-primary",
                  },
                  "ref": null,
                  "rendered": "Pulp Fiction",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1999",
                    "className": "rating",
                  },
                  "ref": null,
                  "rendered": "1999",
                  "type": "span",
                },
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "8.3Thriller,Crime",
                "className": "numbers",
              },
              "ref": null,
              "rendered": "8.3Thriller,Crime",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "154",
                "className": "numbers",
              },
              "ref": null,
              "rendered": "154",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum dolor sit amet.",
                "className": "text-basic",
              },
              "ref": null,
              "rendered": "Lorem ipsum dolor sit amet.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="details"
        >
          <Poster
            path="/img1234_0987"
          />
          <div
            className="info"
          >
            <h2>
              <span
                className="title text-primary"
              >
                Pulp Fiction
              </span>
              <span
                className="rating"
              >
                1999
              </span>
            </h2>
            <p
              className="numbers"
            >
              8.3Thriller,Crime
            </p>
            <p
              className="numbers"
            >
              154
            </p>
            <p
              className="text-basic"
            >
              Lorem ipsum dolor sit amet.
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Poster
              path="/img1234_0987"
            />,
            <div
              className="info"
            >
              <h2>
                <span
                  className="title text-primary"
                >
                  Pulp Fiction
                </span>
                <span
                  className="rating"
                >
                  1999
                </span>
              </h2>
              <p
                className="numbers"
              >
                8.3Thriller,Crime
              </p>
              <p
                className="numbers"
              >
                154
              </p>
              <p
                className="text-basic"
              >
                Lorem ipsum dolor sit amet.
              </p>
            </div>,
          ],
          "id": "details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "path": "/img1234_0987",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  <span
                    className="title text-primary"
                  >
                    Pulp Fiction
                  </span>
                  <span
                    className="rating"
                  >
                    1999
                  </span>
                </h2>,
                <p
                  className="numbers"
                >
                  8.3Thriller,Crime
                </p>,
                <p
                  className="numbers"
                >
                  154
                </p>,
                <p
                  className="text-basic"
                >
                  Lorem ipsum dolor sit amet.
                </p>,
              ],
              "className": "info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="title text-primary"
                    >
                      Pulp Fiction
                    </span>,
                    <span
                      className="rating"
                    >
                      1999
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Pulp Fiction",
                      "className": "title text-primary",
                    },
                    "ref": null,
                    "rendered": "Pulp Fiction",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1999",
                      "className": "rating",
                    },
                    "ref": null,
                    "rendered": "1999",
                    "type": "span",
                  },
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "8.3Thriller,Crime",
                  "className": "numbers",
                },
                "ref": null,
                "rendered": "8.3Thriller,Crime",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "154",
                  "className": "numbers",
                },
                "ref": null,
                "rendered": "154",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem ipsum dolor sit amet.",
                  "className": "text-basic",
                },
                "ref": null,
                "rendered": "Lorem ipsum dolor sit amet.",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PersonDetails renders correctly with fetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonDetails
    details={
      Object {
        "biography": "Lorem ipsum dolor sit amet.",
        "birthday": "8.3",
        "deathday": "154",
        "name": "Pulp Fiction",
        "place_of_birth": Array [
          "Thriller",
          "Crime",
        ],
        "popularity": "1999",
        "profile_path": "/img1234_0987",
      }
    }
    fetching={true}
    getFilmsWithCast={[Function]}
    getPersonDetails={[Function]}
    setSearchType={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Loading
        className=""
        color="#fff"
        delay={0}
        height={64}
        type="bars"
        width={64}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "className": "",
        "color": "#fff",
        "delay": 0,
        "height": 64,
        "type": "bars",
        "width": 64,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Loading
          className=""
          color="#fff"
          delay={0}
          height={64}
          type="bars"
          width={64}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "color": "#fff",
          "delay": 0,
          "height": 64,
          "type": "bars",
          "width": 64,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PersonDetails renders correctly without details 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonDetails
    details={Array []}
    getFilmsWithCast={[Function]}
    getPersonDetails={[Function]}
    setSearchType={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
